inbox/api/ns_api.py

```diff
@@ -625,20 +641,24 @@ def folder_label_delete_api(public_id):
                 "Folder {} cannot be deleted because it contains messages.".
                 format(public_id))
 
-        deleted_at = datetime.utcnow()
-        category.deleted_at = deleted_at
-        folders = category.folders if g.namespace.account.discriminator \
-            != 'easaccount' else category.easfolders
-        for folder in folders:
-            folder.deleted_at = deleted_at
+        if g.api_version == 1:
+            # Optimistically update folders.
+            deleted_at = datetime.utcnow()
+            category.deleted_at = deleted_at
+            folders = category.folders if g.namespace.account.discriminator \
+                != 'easaccount' else category.easfolders
+            for folder in folders:
+                folder.deleted_at = deleted_at
 
         schedule_action('delete_folder', category, g.namespace.id,
                         g.db_session)
     else:
-        deleted_at = datetime.utcnow()
-        category.deleted_at = deleted_at
-        for label in category.labels:
-            label.deleted_at = deleted_at
+        if g.api_version == 1:
```

Comment:
This might be cleaner if we translate the API version into API feature flags, and then check those. The comment (`#Optimistically update labels`) wouldn't be as necessary if we could say `if g.api_features.optimistic_updates:`